x/mongo/driver/topology/connection.go

```diff
@@ -638,7 +638,7 @@ func (c *Connection) CompressWireMessage(src, dst []byte) ([]byte, error) {
 		return dst, ErrConnectionClosed
 	}
 	if c.connection.compressor == wiremessage.CompressorNoOp {
-		return append(dst, src...), nil
+		return src, nil
```

Comment:
To prevent backward-breaking changes and confusing behavior in the future, we should still support the case where `dst` is not empty/nil.

E.g.
```go
if len(dst) == 0 {
	return src, nil
}
return append(dst, src...), nil
```